Draft: Phlex 2 support

triskweline opened the pull request:
This is a draft PR that adds support for Phlex 2.1 (and drops support for Phlex 1).

So far I fixed the tests and all deprecation warnings. I didn't do any testing beyond that.

triskweline committed:
Bump dependencies to phlex 2.1+

triskweline committed:
Fix deprecation warning from engine mount in tests

triskweline committed:
Green tests for Phlex 2

joelmoss commented:
Amazing! Thanks for this, but I don't want to drop support for Phlex 1 just yet. I still use it heavily, and rely a lot on some of the things changed or removed in Phlex 2.

Version detection of Phlex would be best here. Any chance you could add that, so that these changes are only applied when Phlex 2+ is used?

triskweline committed:
Support both Phlex 1 and 2

triskweline commented:
I added version switches for Phlex 1/2.

I also added the `appraisals` gem to run tests for various Phlex versions. Tested gem sets are defined in the `Appraisal` file.

To run tests for all supported Phlex versions:

```
bundle exec appraisal sus
```

To run tests for a specific Phlex versions, call:

```bash
bundle exec appraisal phlex-1 sus
bundle exec appraisal phlex-2 sus
```

Again, I only made sure the tests pass, but did no manual testing beyond that.

joelmoss commented:
Thanks loads. Will take a look.
